fideslog/api/schemas/user_registration_event.py

```diff
@@ -0,0 +1,62 @@
+# pylint: disable= no-self-argument, no-self-use
+
+from datetime import datetime, timezone
+
+from pydantic import BaseModel, Field, validator
+
+
+class UserRegistrationEvent(BaseModel):
+    """The model for user registration events."""
+
+    analytics_id: str = Field(
+        ...,
+        description="The fully anonymized, globally unique identifier for the event sender.",
+    )
+    email: str = Field(
+        ...,
+        description="User registration event email",
+    )
+    organization: str = Field(
+        ...,
+        description="TUser registration event organization",
+    )
+    registered_at: datetime = Field(
+        ...,
+        description="The UTC timestamp when the user registration event occurred, in ISO 8601 format. Must include "
+        "the UTC timezone, and represent a datetime in the past.",
+    )
+
+    @validator("analytics_id")
+    def check_not_an_email_address(cls, value: str) -> str:
+        """
+        Validate that analytics_id does not contain an email address literal.
+        """
+
+        assert value.find("@") == -1, "analytics_id must not be identifiable"
+        return value
+
+    @validator("email")
+    def check_email(cls, value: str) -> str:
+        """
+        Validate that email is an email.
+        """
+
+        assert value.find("@") != -1, "email must contain a valid email address"
+        return value
```

Comment:
Email address validation is a hard problem to solve. Instead of defining the field as a simple `str` and attempting to implement our own validator, it would be best to use `EmailStr`. See [the pydantic docs](https://pydantic-docs.helpmanual.io/usage/types/#pydantic-types) for info on that.